### PR TITLE
Validates config JSON files

### LINKED ISSUE #8: Verify the structure/contents of the config JSON files

### PR DESCRIPTION
**Pull request by wolfie**
Fixes #8

### CODE REVIEW

File: _run.ts [hunk 1]
```diff
@@ -70,22 +69,75 @@ namespace runner {
...
+                ANSI_RESET
+        );
+        errorLines.forEach(line => console.error(ANSI_RED + line + ANSI_RESET));
+        console.error();
+    }
+
+    async function getValidatedJson<T>(
+        jsonFilename: string,
+        schemaFilename: string,
+        noSuchFileCB: () => any
```
Comment (OllesEtta) [inline _run.ts]: What does CB mean in this context?

File: _run.ts [hunk 2]
```diff
@@ -70,22 +69,75 @@ namespace runner {
...
+            const json: T = await import(jsonFilename);
+
+            // TS creates the default property automatically, but
+            // 1) we don't need it, and
+            // 2) it messes up the validation
+            delete (<any>json).default;
+
+            const validatorResult = validateJsonSchema(json, schema);
+            if (validatorResult.errors.length === 0) {
+                return JSON.parse(JSON.stringify(json)); // return copy
```
Comment (OllesEtta) [inline _run.ts]: Comment could also explain why we want to return copy.

File: _run.ts (hunk 1 above)
Comment (OllesEtta) [reply _run.ts]: Answering to myself, probably callback. Maybe use Callback instead of CB so that is more clear for reader?

Comment (wolfie) [reply _run.ts]: "cb" is such a common abbreviation in TypeScript/JavaScript that I'll let it be this time

**Pull request merged**